### ISSUE TITLE
[patch] make.bat improve error message if Sphinx is not found

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1024]
Attached patch guides Windows users what to do if sphinx-build is not found.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1024
- Originally reported by: anatoly techtonik
- Originally created at: 2012-10-22T17:47:43.856

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-03 09:21:08+00:00_

The patch is fair enough. I think if you make this change, you may have to change the Makefile is good as well.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-11-03 09:57:01+00:00_

With some grammar fixes in the text, good to go in:
-    echo.The 'sphinx-build' command was not found. Make sure you have Sphinx
-    echo.installed, then set the SPHINXBUILD environment variable to point
-    echo.to the full path of the 'sphinx-build' executable. Alternatively you
-    echo.may add the Sphinx directory to PATH.

**Comment by shimizukawa** [comment]
_From anatoly techtonik on 2012-11-03 16:23:20+00:00_

Fixed and tested patch - see pull request #84.

**Comment by shimizukawa** [comment]
_From anatoly techtonik on 2012-11-03 16:27:56+00:00_

I don't know Linux error codes to patch Makefile, and the text should be different as Linux distributions have their own obscure ways of installing Python packages.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-07 10:26:55+00:00_

Closes #1024: Makefile improve error message if Sphinx is not found

→ <<cset dd09c4ce45f2971a30e9abb17f2bd7cde1554261>>

**Comment by shimizukawa** [comment]
_From anatoly techtonik on 2012-11-07 10:31:56+00:00_

Are you sure there is the PATH variable on the platforms that Makefile is destined to run on? I thought it is Windows-only feature.

**Comment by shimizukawa** [comment]
_From Jon Waltman on 2012-11-08 05:35:59+00:00_

`PATH` is the standard variable on Unix-like platforms as well.

http://www.linfo.org/path_env_var.html

**Comment by shimizukawa** [comment]
_From anatoly techtonik on 2012-11-08 08:18:53+00:00_

Thanks for the info. I didn't know that.

**Comment by shimizukawa** [comment]
_From Jon Waltman on 2012-11-08 10:41:32+00:00_

This breaks the Makefile on Linux because the check for sphinx-build always fail.

```
% ls
_build  conf.py  index.rst  make.bat  Makefile  _static  _templates
% sphinx-build
Sphinx v1.2pre
Usage: /home/jon/Desktop/s1/e/bin/sphinx-build [options] sourcedir outdir [filenames...]
Options: -b <builder> -- builder to use; default is html
         -a        -- write all files; default is to only write new and changed files
         -E        -- don't use a saved environment, always read all files
         -t <tag>  -- include "only" blocks with <tag>
         -d <path> -- path for the cached environment and doctree files
                      (default: outdir/.doctrees)
         -c <path> -- path where configuration file (conf.py) is located
                      (default: same as sourcedir)
         -C        -- use no config file at all, only -D options
         -D <setting=value> -- override a setting in configuration
         -A <name=value>    -- pass a value into the templates, for HTML builder
         -n        -- nit-picky mode, warn about all missing references
         -N        -- do not do colored output
         -q        -- no output on stdout, just warnings on stderr
         -Q        -- no output at all, not even warnings
         -w <file> -- write warnings (and errors) to given file
         -W        -- turn warnings into errors
         -P        -- run Pdb on exception
Modi:
* without -a and without filenames, write new and changed files.
* with -a, write all files.
* with filenames, write these.
% echo $?                                                                                                                                          [1]
1
% make html
Makefile:22: *** 

The 'sphinx-build' command was not found. Make sure you have Sphinx
installed, then set the SPHINXBUILD environment variable to point
to the full path of the 'sphinx-build' executable. Alternatively you
may add the Sphinx directory to PATH.

If you don't have Sphinx installed, grab it from
http://sphinx-doc.org/ .  Stop.
% 
```

Running sphinx-build with no arguments exits with non-zero status code.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-08 22:44:16+00:00_

[Jon Waltman](https://bitbucket.org/jonwaltman) Sorry, It's my fault. My buildout(bit old, newest version fix this issue) environment's sphinx-build command always return 0.

http://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html says "command not found" return `127`.

I'll fix this soon.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-08 22:54:24+00:00_

Closes #1024: 'command not found' status code is 127.

→ <<cset 22d1280dac44f29f816ea4327d27e62e81eb78b0>>
